File: java/src/test/java/com/searchio/interview/fetcher/TestEntity.java
package com.searchio.interview.fetcher;

import com.searchio.interview.model.Entity;

import java.util.Objects;

final class TestEntity implements Entity {
    private final Long id;
    private final Object source;

    public TestEntity(Long id, Object source) {
        this.id = id;
        this.source = source;
    }

    @Override
    public long getIdentifier() {
        return id;
    }

    @Override
    public Object getSource() {
        return source;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        TestEntity that = (TestEntity) o;
        return Objects.equals(id, that.id) && Objects.equals(source, that.source);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, source);
    }
}

File: csharp/SearchIOCodingExercise/TestCachingExercise/TestEntity.cs
using System;
using CachingExercise.Model;

namespace TestCachingExercise
{
    public class TestEntity : IEntity
    {
        private long _id;
        private Object _source;

        public TestEntity(long id, object source)
        {
            this._id = id;
            this._source = source;
        }

        public long GetIdentifier()
        {
            return this._id;
        }

        public object GetSource()
        {
            return this._source;
        }

        protected bool Equals(TestEntity other)
        {
            return _id == other._id && Equals(_source, other._source);
        }

        public override bool Equals(object obj)
        {
            if (ReferenceEquals(null, obj)) return false;
            if (ReferenceEquals(this, obj)) return true;
            if (obj.GetType() != this.GetType()) return false;
            return Equals((TestEntity) obj);
        }

        public override int GetHashCode()
        {
            unchecked
            {
                return (_id.GetHashCode() * 397) ^ (_source != null ? _source.GetHashCode() : 0);
            }
        }
    }
}